raiden-ts/src/transport/epics/presence.ts

```diff
@@ -209,28 +209,32 @@ export function matrixPresenceUpdateEpic(
       // filter out events from users we don't care about
       // i.e.: presence monitoring never requested
       filter(([{ address }, toMonitor]) => toMonitor.has(address)),
-      mergeMap(([{ matrix, user, address }, , { pollingInterval, httpTimeout }]) => {
-        // first filter can't tell typescript this property will always be set!
-        const userId = user.userId,
-          presence = user.presence!,
-          available = AVAILABLE.includes(presence);
-
-        // always fetch profile info, to get up-to-date displayname & avatar_url
-        return defer(() => matrix.getProfileInfo(userId)).pipe(
-          map((profile) => {
+      mergeMap(([{ matrix, user, address }, , { pollingInterval, httpTimeout }]) =>
+        defer(async () =>
```

Comment:
I guess you should use `retryAsync$` here to handle matrix rate-limit erros

Reply:
It is handled, but using `retryWaitWhile` a little down this observable chain

Reply:
Missed that one, thanks!